examples/frontend/containers/Layout.tsx

```diff
@@ -54,11 +60,15 @@ export default function Layout({ children, project = 'share' }) {
 
     const preSignOut = () => {
         localStorage.removeItem('contractAddress')
-        signOut()
         router.reload(window.location.origin + router.asPath)
+        signOut()
     }
 
     useEffect(switchProject, [router.asPath])
+    useEffect(switchProject, [])
+    useEffect(()=>{
+        setSignedProject(localStorage.getItem('contractAddress'))
```

Comment:
Wouldn't be better if we move this line into `switchProject` function?

Reply:
I just wanted to run it only when rendering the page. Anyway it won't be an issue if we move it to the `switchProject` function.